src/backend/aspen/api/views/phylo_trees.py

```diff
@@ -0,0 +1,51 @@
+import sentry_sdk
+import sqlalchemy as sa
+from fastapi import APIRouter, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import aliased, joinedload
+from sqlalchemy.sql.selectable import Select
+
+from aspen.api.auth import get_auth_user
+from aspen.api.deps import get_db
+from aspen.api.error import http_exceptions as ex
+from aspen.api.schemas.phylo_trees import PhyloTreeRequest, PhyloTreeResponse
+from aspen.database.models import Group, PhyloRun, PhyloTree, User
+
+router = APIRouter()
+
+
+@router.put("/")
+async def update_phylo_tree(
+    phylo_tree_request: PhyloTreeRequest,
+    db: AsyncSession = Depends(get_db),
+    user: User = Depends(get_auth_user),
+):
+    phylo_run_alias = aliased(PhyloRun)
+
+    # use associated PhyloRun to check that user has permission to update PhyloTree
+    phylo_runs_query: Select = (
+        sa.select(phylo_run_alias)  # type: ignore
+        .options(
+            joinedload(phylo_run_alias.outputs.of_type(PhyloTree)),
+            joinedload(phylo_run_alias.group),
+        )
+        .filter((PhyloTree.id == phylo_tree_request.id))
+    )
+    phylo_run_results = await db.execute(phylo_runs_query)
+    phylo_run: PhyloRun = phylo_run_results.unique().scalars().one()
+
+    group: Group = user.group
+    if phylo_run.group_id != group.id:
+        if not user.system_admin:
+            not_sufficient_permission_msg: str = f"User {user.name} from group {group.name} does not have permission to update tree name"
+            sentry_sdk.capture_message(not_sufficient_permission_msg, "error")
+            raise ex.BadRequestException(not_sufficient_permission_msg)
+
+    await db.execute(
+        sa.update(PhyloTree)  # type: ignore
+        .where(PhyloTree.id == phylo_tree_request.id)
+        .values(name=phylo_tree_request.name)
+    )
```

Comment:
I usually prefer using the ORM interface to update small numbers of rows, something more like:

```
tree_obj = phylo_run.outputs[0]
tree_obj.name = phylo_tree_request.name
await db.commit()
```

Reply:
oh really? I thought that this was the correct way to update in [sqlalchemy 2.0]( https://docs.sqlalchemy.org/en/14/orm/session_basics.html#orm-expression-update-delete)

Reply:
That's the correct way to issue an explicit `UPDATE` query - which is a capability we definitely need if we want to update more than a handful of rows at once, but if we already have ORM objects loaded and aren't trying to update many rows at once, I think this is the section of the docs we want to look at:
 https://docs.sqlalchemy.org/en/14/tutorial/orm_data_manipulation.html#updating-orm-objects